test/verifiedTests/testMetaboToolsTutorialI/testMetaboTools_tutorialI.m

```diff
@@ -0,0 +1,371 @@
+        
+        load('testdata_setQualitativeConstraintsB.mat');
+        [model_B] = setQualitativeConstraints(modelMedium, cond2_uptake, cond2_uptake_LODs, cond2_secretion, cond2_secretion_LODs, cellConc, t, cellWeight, ambiguous_metabolites, basisMedium);
+        
+        % % % test on output1
+        FBAA = optimizeCbModel(model_A);
+        FBAB = optimizeCbModel(model_B);
+        
+        % % % % test on modelMedium
+        ovA =0.9228;
+        ovB =0.9228;
+        assert(FBAA.f - ovA < toltests); % tol must be defined previously, e.g. tol = 1e-6;
+        assert(FBAB.f - ovB < toltests); % t
+                
+        
+        %% Step *5* setSemiQuantConstraints
+        % This function applies the constraints to the models. It takes two
+        % condition specific models into consideration.
+        
+        toltests = 1.0000e-04; % reduce for FBA results.
+        [modelA_QUANT, modelB_QUANT] = setSemiQuantConstraints(model_A, model_B, cond1_upt_higher, cond2_upt_higher, cond2_secr_higher, cond1_secr_higher);
+        
+        FBAA = optimizeCbModel(modelA_QUANT);
+        FBAB = optimizeCbModel(modelB_QUANT);
+        
+        % % % % test on setSemiQuantConstraints
+        ovA =0.9228;
+        ovB =0.7227;
+        assert(FBAA.f - ovA < toltests);
+        assert(abs(FBAB.f - ovB) < toltests); %
+        
+        
+        
+        %% Step *6* Apply growth constraints
+        
+        load('testdata_setConstraintsOnBiomassReactionA');
+        [model_A_BM] = setConstraintsOnBiomassReaction(modelA_QUANT, of, dT, tolerance);
+        
+        
+        %%
+        % set constraints on CCRF-CEM model
+        load('testdata_setConstraintsOnBiomassReactionB');
+        
+        [model_B_BM] = setConstraintsOnBiomassReaction(modelB_QUANT, of, dT, tolerance);
+        
+        
+        FBAA = optimizeCbModel(model_A_BM);
+        FBAB = optimizeCbModel(model_B_BM);
+        
+        % % % % test on setSemiQuantConstraints
+        ovA =0.0424;
+        ovB =0.0378;
+        assert(FBAA.f - ovA < toltests);
+        assert(FBAB.f - ovB < toltests); %
+        
+        %% Step *7* integrateGeneExpressionData
+        
+        dataGenes = [535
+            1548
+            2591
+            3037
+            4248
+            4709
+            6522
+            7167
+            7367
+            8399
+            23545
+            129807
+            221823
+            ];
+        
+        [model_A_GE] = integrateGeneExpressionData(model_A_BM, dataGenes);
+        
+        
+        dataGenes = [239
+            443
+            535
+            1548
+            2683
+            3037
+            4248
+            4709
+            5232
+            6522
+            7364
+            7367
+            8399
+            23545
+            54363
+            66002
+            129807
+            221823
+            ];
+        
+        [model_B_GE] = integrateGeneExpressionData(model_B_BM, dataGenes);
```

Comment:
could you cherry-pick my commit on https://github.com/tpfau/cobratoolbox/commit/13b5128df01ca0aa2773ddf22426464b1156824a that actually allows using non-numeric gene sets to be used in integrateGeneExpressionData ?
It still works with the old numeric stuff from Recon1.